case-lib/lib.sh

```diff
@@ -264,3 +264,12 @@ sudo sync -f || true
 if [ ! "$DMESG_LOG_START_LINE" ]; then
     DMESG_LOG_START_LINE=$(wc -l /var/log/kern.log|awk '{print $1;}')
 fi
+
+is_sof_platform()
```

Comment:
this should be is_hda_legacy_platform()

and have additional checks for the PCI ID.

Reply:
I think we should make this as is_sof_used or is_hda_legacy_used as we can switch to use both on same platform.

Reply:
now I use /proc/aound/cards to tell if we are using SOF, if yes, dump pipelines from tplg, if not, dump pipelines from proc, it's much more dependable than just check the kernel module.